src/main/java/rx/internal/operators/OperatorEagerConcatMap.java

```diff
@@ -220,6 +223,7 @@ void drain() {
                             if (empty) {
                                 synchronized (subscribers) {
                                     subscribers.poll();
+                                    request(1);
```

Comment:
Please call `request(1)` outside the synchronized block and preferrably after `innerDone = true`.